src/compiler/preprocess/index.ts

```diff
@@ -42,17 +42,19 @@ interface Replacement {
 
 async function replace_async(str: string, re: RegExp, func: (...any) => Promise<string>) {
 	const replacements: Array<Promise<Replacement>> = [];
-	str.replace(re, (...args) => {
-		replacements.push(
-			func(...args).then(
-				res => // eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
-					({
-						offset: args[args.length - 2],
-						length: args[0].length,
-						replacement: res,
-					}) as Replacement
-			)
-		);
+	str.replace(re, (match, attributes, content, offset) => {
```

Comment:
Were you just making this change for code style reasons? Part of the reason it was written like it was was so that replace_async would work with arbitrarily many capture groups. With these changes, it looks like anything but the first capture group will get discarded and not passed on to the callback.